### PR TITLE
Bootstrap service

### PR DESCRIPTION
**Pull request by yurivin**
- Btc Wallet creation service
- Startup configuration in ReadMe.md
- Additional adjustments

### CODE REVIEW

File: bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        val wallet = Wallet.fromSeed(netParams, seed)
+
+        val out = ByteArrayOutputStream(1024)
+        wallet.saveToFileStream(out)
+        val walletBytes = out.toByteArray()
+        out.reset()
+        return BtcWallet(DatatypeConverter.printBase64Binary(walletBytes), network)
+    }
+
+    private fun createBtcWallet(netParams: NetworkParameters): BtcWallet {
```
Comment (dolgopolovwork) [inline bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt]: This method is not used

File: bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            log.error("Error creating Bitcoin wallet", e)
+            val response = BtcWallet(network = network)
+            response.errorCode = e.javaClass.simpleName
+            response.message = e.message
+            return ResponseEntity.ok<BtcWallet>(response)
+        }
+    }
+
+    private fun createWalletVer2(netParams: NetworkParameters): BtcWallet {
+        val seedBytes = Random.nextBytes(DeterministicSeed.DEFAULT_SEED_ENTROPY_BITS / 8)
```
Comment (dolgopolovwork) [inline bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt]: I don't think we need this. `Wallet(netParams)` creates random seed for you. Right from javadoc:

> Creates a new, empty wallet with a randomly chosen seed and no transactions.

File: bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        val mnemonic = MnemonicCode.INSTANCE.toMnemonic(seedBytes)
+        val passphrase = ""
+        val creationTimeSeconds = System.currentTimeMillis() / 1000
+
+        val seed = DeterministicSeed(mnemonic, null, passphrase, creationTimeSeconds)
+        val wallet = Wallet.fromSeed(netParams, seed)
+
+        val out = ByteArrayOutputStream(1024)
```
Comment (dolgopolovwork) [inline bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt]: Shall we close this stream?

File: bootstrap/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# TestNet3, MainNet, RegTest
+btc.network=RegTest
```
Comment (dolgopolovwork) [inline bootstrap/src/main/resources/application.properties]: I'd rather create 3 endpoints in 3 different modes. I think it's more convenient.

File: bootstrap/src/test/kotlin/jp/co/soramitsu/bootstrap/research/BtcResearch.kt [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+import org.json.JSONObject
+import org.web3j.crypto.*
+import java.util.*
+import kotlin.random.Random
+
+
+
+
+
+class BtcResearch {
```
Comment (dolgopolovwork) [inline bootstrap/src/test/kotlin/jp/co/soramitsu/bootstrap/research/BtcResearch.kt]: t think this file must be removed.

File: bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt (hunk 3 above)
Comment (yurivin) [reply bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt]: If you'll read java doc for ByteArrayOutputStream
/**
     * Closing a <tt>ByteArrayOutputStream</tt> has no effect. The methods in
     * this class can be called after the stream has been closed without
     * generating an <tt>IOException</tt>.
     */

File: bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt (hunk 1 above)
Comment (yurivin) [reply bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt]: fixed

File: bootstrap/src/test/kotlin/jp/co/soramitsu/bootstrap/research/BtcResearch.kt (hunk 5 above)
Comment (yurivin) [reply bootstrap/src/test/kotlin/jp/co/soramitsu/bootstrap/research/BtcResearch.kt]: done

File: bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt (hunk 3 above)
Comment (dolgopolovwork) [reply bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt]: well ok then

File: bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt (hunk 2 above)
Comment (yurivin) [reply bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/BtcController.kt]: Done

File: bootstrap/src/main/resources/application.properties (hunk 4 above)
Comment (yurivin) [reply bootstrap/src/main/resources/application.properties]: Looks like BitcoinJ does not support multiple networks in one application instance
Here how to test
  @Test
    fun testBitcoinJWalletNetworks() {

        val wallet = Wallet(RegTestParams.get())

        val wallet2 = Wallet(TestNet3Params.get())
    }

Got:
java.lang.IllegalStateException: Context does not match implicit network params: org.bitcoinj.params.RegTestParams@2b55c76b vs org.bitcoinj.params.TestNet3Params@14ccf24b

File: bootstrap/build.gradle [hunk 6]
```diff
@@ -31,8 +31,7 @@ dependencies {
     testCompile "org.jetbrains.kotlin:kotlin-test:$kotlinVersion"
 
     //Iroha-java related
-    compile 'com.github.warchant:iroha-pure-java:v1.0.0_rc2-SNAPSHOT'
-    implementation('com.github.warchant:testcontainers-iroha:0.2.1') {
+    implementation('com.github.hyperledger.iroha-java:testcontainers:2.0.0') {
```
Comment (Mingela) [inline bootstrap/build.gradle]: since we use 1.0.0_rc4-hotfix1 now it should be changed to `3.1.0`

Comment (yurivin) [reply bootstrap/build.gradle]: fixed

**Pull request merged**